cli/docker/build.go

```diff
@@ -80,6 +77,31 @@ Loop:
 	return nil
 }
 
+func convertDockerOutputToString(outputMap map[string]interface{}) string {
+	// The data format is a bit strange: either it is completely
+	// stored in the "stream" field, or it needs to be collected
+	// piece by piece from the rest of the map.
+	if outputMap["stream"] != nil {
+		return outputMap["stream"].(string)
+	}
+
+	// Any of these fields can be null.
+	var output string
+	if outputMap["id"] != nil {
+		output = fmt.Sprintf("%s: ", outputMap["id"])
+	}
+
+	if outputMap["status"] != nil {
+		output = fmt.Sprintf("%s%s ", output, outputMap["status"])
+	}
+
+	if outputMap["progress"] != nil {
+		output = fmt.Sprintf("%s%s", output, outputMap["progress"])
+	}
+
+	return fmt.Sprintf("%s\n", output)
+}
```

Comment:
Everything here looks strange for me. Maybe it is actually the strage API, I don't know. I don't find an information about the format of the return value. I'll ask several questions.

1. We ignore everything if there is the 'stream' field. Is it okay?
2. We don't include keys into the output (we output `<id>: `, not `id: <id>`). Is it okay?
3. We output all fields in different ways (`<id>: `, `<status> `, `<progress>`). Those fields form a single line?
4. It seems, we read the whole json response, so why not to just dump it as is (or dump with prettifying)? Maybe except 'stream', which, I guess, large and would be better to output separately.
5. You said that the response format has different variants. When you see one format and when another? From what it depends? Docker server version? Docker client version? Something else?

Anyway, if it gives more readable result, I don't mind.

Reply:
I made the print the same as in the ``docker pull`` command.

1. Yes. For some reason (I don't know why) the entire output is passed either in the "stream" field, or it needs to be assembled piece by piece from other fields.
2. Yes.
3. Yes. 
4. Because there are fields with information that we do not need (at least, it is not on the server when we do a docker pull)
5. Depends on the action being performed on the docker side. I do not know why.

Reply:
I meet Alexey and now I got the logic. In brief:

* I missed that we process a stream of JSON responses (in a cycle), not just one JSON response.
* So now I understand: responses from `docker pull` have no 'stream' fied, so we process them differently.
* Those `docker pull` responses may or may not appear depending on whether the local image is the newest one.

Thank you for the clarification!